crates/pop-parachains/src/call/mod.rs

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::errors::Error;
+use pop_common::create_signer;
+use subxt::{
+	dynamic::Value,
+	tx::{DynamicPayload, Payload},
+	OnlineClient, SubstrateConfig,
+};
+
+pub mod metadata;
+
+/// Sets up an OnlineClient instance for connecting to a blockchain.
+///
+/// # Arguments
+/// * `url` - Endpoint of the node.
+pub async fn set_up_client(url: &str) -> Result<OnlineClient<SubstrateConfig>, Error> {
+	OnlineClient::<SubstrateConfig>::from_url(url)
+		.await
+		.map_err(|e| Error::ConnectionFailure(e.to_string()))
+}
+
+/// Constructs a dynamic extrinsic payload for a specified pallet and extrinsic.
+///
+/// # Arguments
+/// * `pallet_name` - The name of the pallet containing the extrinsic.
+/// * `extrinsic_name` - The specific extrinsic name within the pallet.
+/// * `args` - A vector of string arguments to be passed to the extrinsic.
+pub async fn construct_extrinsic(
+	pallet_name: &str,
+	extrinsic_name: &str,
+	args: Vec<String>,
+) -> Result<DynamicPayload, Error> {
+	let parsed_args: Vec<Value> = metadata::parse_extrinsic_arguments(args).await?;
+	Ok(subxt::dynamic::tx(pallet_name, extrinsic_name, parsed_args))
+}
+
+/// Signs and submits a given extrinsic to the blockchain.
+///
+/// # Arguments
+/// * `client` - Reference to an `OnlineClient` connected to the chain.
+/// * `tx` - The transaction to be signed and submitted.
+/// * `suri` - The secret URI (e.g., mnemonic or private key) for signing the extrinsic.
+pub async fn sign_and_submit_extrinsic(
```

Comment:
Not tested

Reply:
A successfully `submit_extrinsic` test is tricky because it requires launching a local chain to perform the test unless we find a way to mock the chain interactions effectively.

However, you are right, it feels week not having a test. I have added a simple test to ensure at least that an incorrect transaction fails as expected: https://github.com/r0gue-io/pop-cli/pull/316/commits/0d2551adb4958e4589b50f5391d362193fc22d1e

Reply:
Thanks for adding!